src/ptests/test_unstruc_quad_f90.F90

```diff
@@ -0,0 +1,362 @@
+  end = start + 3*comm_size - 1
+  CALL cgp_section_write_f(F, B, Z, 'Bottom', CGNS_ENUMV(BAR_2), start, end, 0, S, ierr)
+  IF (ierr .NE. CG_OK) CALL cgp_error_exit_f
+
+! Parent elements/side data 
+  DO k=1,3
+    elements(3*0+k) = comm_rank*3+k  ! Element
+    elements(3*1+k) = 0
+    elements(3*2+k) = 1 ! Side
+    elements(3*3+k) = 0
+  ENDDO
+
+  emin(1) = comm_rank*3+start
+  emax(1) = emin(1)+2
+  PRINT *, comm_rank, ':', emin(1), ' ', emax(1) 
+  
+  CALL cgp_parent_data_write_f(F, B, Z, S, emin, emax, elements, ierr)
+  IF (ierr .NE. CG_OK) CALL cgp_error_exit_f
+
+! side connectivity
+  DO k=1,3
+    elements(2*(k-1)+1) = 2*(comm_rank*3+(k-1))+1
+    elements(2*(k-1)+2) = 2*(comm_rank*3+(k-1))+3
+  ENDDO
+  CALL cgp_elements_write_data_f(F, B, Z, S, emin(1), emax(1), elements, ierr)
+  IF (ierr .NE. CG_OK) CALL cgp_error_exit_f
+
+  n_boco_elems = end-start+1
+
+  CALL cg_boco_write_f(F, B, Z, "BottomBC", CGNS_ENUMV(FamilySpecified), CGNS_ENUMV(PointList), n_boco_elems, null_ptr, BC, ierr)
+  IF (ierr .NE. CG_OK) CALL cgp_error_exit_f
+
+  i = 1
+  DO k=emin(1),emax(1)
+    elements(i) = k
+    i = i + 1
+  ENDDO
+
+  start_local(1) = comm_rank * 3 + 1
+  end_local(1) = start_local(1) + 2
+  PRINT *, comm_rank, ':', start_local(1), ' ', end_local(1) 
+
+  indices(1) = Z
+  indices(2) = 1
+  indices(3) = BC
+  indices(4) = 0
+
+  ierr = cg_golist(F, B, depth, pt_labels, indices)
+  IF (ierr .NE. CG_OK) CALL cgp_error_exit_f
+  PRINT *, elements(1), elements(2), start_local(1), end_local(1)
+  CALL cgp_ptlist_write_data_f(F, start_local, end_local, elements, ierr)
+  IF (ierr .NE. CG_OK) CALL cgp_error_exit_f
+
+  CALL cg_boco_gridlocation_write_f(F, B, Z, BC, CGNS_ENUMV(EdgeCenter), ierr)
+  IF (ierr .NE. CG_OK) CALL cgp_error_exit_f
+
+! Add family tag
+  ierr =  cg_golist(F, B, 3, pt_labels, indices)
+  IF (ierr .NE. CG_OK) CALL cgp_error_exit_f
+
+  CALL cg_famname_write_f("Bottom", ierr)
+  IF (ierr .NE. CG_OK) CALL cgp_error_exit_f
+  
+
+! left BC
+  start = end + 1
+  end = start
+
+  CALL cgp_section_write_f(F, B, Z, 'Left', CGNS_ENUMV(BAR_2), start, end, 0, S, ierr)
+  IF (ierr .NE. CG_OK) CALL cgp_error_exit_f
+
+  IF (comm_rank .EQ. 0) THEN
+    emin(1) = start
+    emax(1) = end
+! Parent Element/Side data
+    elements(1) = 1 ! Element
+    elements(2) = 0
+    elements(3) = 4 ! Side
+    elements(4) = 0
+    el_ptr => elements
+  ELSE
+    emin(1) = 0
+    emax(1) = 0
+    el_ptr => NULL()
+  ENDIF
+  PRINT *, comm_rank, ":", emin(1), " ", emax(1)
+
+  CALL cgp_parent_data_write_f(F, B, Z, S, emin, emax, el_ptr, ierr)
+  IF (ierr .NE. CG_OK) CALL cgp_error_exit_f
+
+  IF (comm_rank .EQ. 0) THEN
+    elements(1) = 1
+    elements(2) = 2
+  ENDIF
+  
+  CALL cgp_elements_write_data_f(F, B, Z, S, emin(1), emax(1), el_ptr, ierr)
+  IF (ierr .NE. CG_OK) CALL cgp_error_exit_f
+
+  n_boco_elems = 1
+  CALL cg_boco_write_f(F, B, Z, "LeftBC", CGNS_ENUMV(FamilySpecified), CGNS_ENUMV(PointList), n_boco_elems, null_ptr, BC, ierr)
```

Comment:
can you use c_null_ptr instead to make sure that works?

Reply:
it does not work with c_null_ptr

Reply:
I think you can also do:
integer(cgsize), pointer :: c_ptr
call C_F_POINTER(C_NULL_PTR, c_ptr)
and pass c_ptr.

It might need to be an array, though
integer(cgsize), pointer :: c_ptr(:)
call C_F_POINTER(C_NULL_PTR, c_ptr, [0])

I need to become more familiar with null() in Fortran. I'm surprised it works, given that the argument does not match the API interface.

Reply:
As you said we could have an interface with pure iso_c_binding like what is done for *_f03 subroutines.

But I think that it should be a dedicated task separate from this thread in order to clean the whole fortran interface.

For instance the binding to function that ends with "_f" looks confusing to me and should be better with a '_c'

Reply:
Agreed, I am happy with the PR as is. Are you done with your changes?

Reply:
Yes, I will stop with changes now as  I happy with the current status too.